wasatch/MockUSBDevice.py

```diff
@@ -0,0 +1,312 @@
+        self.fake_pid = str(hash(spec_name))
+        self.device_id = DeviceID(label=f"USB:{self.fake_pid[:8]}:0x16384:111111:111111")
+        self.device_id = self.device_id
+        self.bus = self.device_id.bus
+        self.address = self.device_id.address
+        self.vid = self.device_id.vid
+        self.pid = self.device_id.pid
+
+        #path attributes
+        self.test_spec_dir = os.path.join(self.get_default_data_dir(), 'testSpectrometers')
+        self.spectrometer_folder = self.get_spec_folder()
+        self.test_spec_readings = os.path.join(self.test_spec_dir, self.spectrometer_folder,'readings')
+        self.test_spec_eeprom = os.path.join(self.test_spec_dir, self.spectrometer_folder,'eeprom')
+
+        #init attributes
+        self.spec_readings = {}
+        self.int_time = 1000
+        self.detector_gain = 10
+        self.detector_offset = 1
+        self.detector_setpoint = 1
+        self.detector_temp_raw = 40.0
+        self.disconnect = False
+        self.single_reading = False
+        self.got_start_int = False
+        self.laser_enable = False
+        self.got_start_detector_gain = False
+        self.got_start_detector_offset = False
+        self.got_start_detector_setpoint = False
+        self.detector_tec_enable = False
+
+        #set up functions
+        self.re_pattern_1 = re.compile('(.)([A-Z][a-z]+)')
+        self.re_pattern_2 = re.compile('([a-z0-9])([A-Z])')
+        self.wpsc_translate = {
+            "wavecal_coeffs":"wavelength_coeffs",
+            "temp_to_dac_coeffs":"degC_to_dac_coeffs",
+            "adc_to_temp_coeffs":"adc_to_degC_coeffs",
+            "serial": "serial_number",
+            "inc_laser": "has_laser",
+            "inc_battery": "has_battery",
+            "inc_cooling": "has_cooling",
+            "max_laser_power_mw": "max_laser_power_mW",
+            "excitation_wavelength_nm": "excitation_nm",
+            "detector_name": "detector",
+            "flip_x_axis": "invert_x_axis",
+            }
+
+        self.load_readings()
+        self.load_eeprom(self.test_spec_eeprom)
+        self.covnert_eeprom()
+        self.reading_len = len(self.spec_readings)
+        self.default_ctrl_return = [1 for i in range(64)]
+        if self.eeprom_overrides:
+            self.override_eeprom()
+        # style is (bRequest,wValue) to allow for second tier op codes
+        # if first tier, where wValue matters then wValue should be given as None
+        self.cmd_dict = {
+            (178,None): self.cmd_set_int_time,
+            (182,None): self.cmd_set_offset,
+            (183,None): self.cmd_set_gain,
+            (190,None): self.cmd_toggle_laser,
+            (214,None): self.cmd_toggle_tec,
+            (215,None): self.cmd_get_detector_temp,
+            (216,None): self.cmd_set_setpoint,
+            (218,None): self.cmd_get_tec_enable,
+            (226,None): self.cmd_get_laser_enabled,
+            (255,1): self.cmd_read_eeprom,
+            }
```

Comment:
I would prefer these be in hex.  It makes it a lot easier when grepping source code for usage of opcodes to have them consistently formatted.